packages/cache-ethereum-storage/src/cache-ethereum-storage.ts

```diff
@@ -0,0 +1,210 @@
+import { LogTypes, StorageTypes } from '@requestnetwork/types';
+import Utils from '@requestnetwork/utils';
+import * as Bluebird from 'bluebird';
+import { EventEmitter } from 'events';
+
+/**
+ * CacheEthereumStorage
+ * @notice Manages storage layer of the Request Network Protocol v2
+ */
+export default class CacheEthereumStorage implements StorageTypes.IStorage {
+  /**
+   * Logger instance
+   */
+  private logger: LogTypes.ILogger;
+
+  // TODO replace by database, files or something...
+  private data: { [id: string]: { content: string; timestamp: number } | null };
+
+  /**
+   * Url where can be reached the data hosted by this storage
+   */
+  private locationUrl: string;
+
+  /**
+   * Url where can be reached the data hosted by this storage
+   */
+  private ethereumStorage: StorageTypes.IStorage;
+
+  private isInitialized: boolean = false;
+
+  /**
+   * Constructor
+   */
+  public constructor(
+    ethereumStorage: StorageTypes.IStorage,
+    locationUrl: string,
+    logger?: LogTypes.ILogger,
+  ) {
+    this.ethereumStorage = ethereumStorage;
+    this.locationUrl = locationUrl;
+    this.logger = logger || new Utils.SimpleLogger();
+    this.data = {};
+  }
+
+  /**
+   * Initialize the module. Does nothing, exists only to implement IStorage
+   *
+   * @returns nothing
+   */
+  public async initialize(): Promise<void> {
+    // initialize storage
+    await this.ethereumStorage.initialize();
+
+    this.isInitialized = true;
+  }
+
+  /**
+   * Append content into the storage: add the content to ipfs and the hash on Ethereum
+   * @param content Content to add into the storage
+   * @returns Promise resolving id used to retrieve the content
+   */
+  public async append(content: string): Promise<StorageTypes.IAppendResult> {
+    if (!this.isInitialized) {
+      throw new Error('Ethereum storage must be initialized');
+    }
+
+    if (!content) {
+      throw Error('No content provided');
+    }
+
+    const result: StorageTypes.IAppendResult = Object.assign(new EventEmitter(), {
+      content,
+      // will be set up later
+      id: '',
+      meta: {
+        // will be set up later
+        ipfs: { size: 0 },
+        storageType: StorageTypes.StorageSystemType.ETHEREUM_IPFS,
+        timestamp: Utils.getCurrentTimestampInSecond(),
+      },
+    });
+
+    const resultAppend = await this.ethereumStorage.append(content);
+
+    resultAppend.on('confirmed', (resultAppendConfirmed: StorageTypes.IAppendResult) => {
+      // Remove cache
+      this.data[resultAppend.id] = null;
+
+      result.emit('confirmed', resultAppendConfirmed);
```

Comment:
would be nice to have some logs about the cache, like one here for example.

Reply:
:+1: